crates/extensions/c8y_http_proxy/src/actor.rs

```diff
@@ -90,17 +90,17 @@ impl Actor for C8YHttpProxyActor {
                     .await
                     .map(|response| response.into()),
 
-                C8YRestRequest::C8yUpdateSoftwareListResponse(request) => self
-                    .send_software_list_http(request.source.clone(), request)
+                C8YRestRequest::SoftwareListResponse(request) => self
+                    .send_software_list_http(request.device_id.clone(), request)
```

Comment:
```suggestion
                    .send_software_list_http(request)
```
When the `request` instance already has the `device_id` in it, why extract it and pass it separately?

Reply:
fixed